fastly/resource_fastly_service_v1_test.go

```diff
@@ -349,6 +349,62 @@ func TestAccFastlyServiceV1_updateInvalidBackend(t *testing.T) {
 	})
 }
 
+func TestAccFastlyServiceV1_createServiceWithStaticBackend(t *testing.T) {
+	var service gofastly.ServiceDetail
+	name := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+	domain := fmt.Sprintf("fastly-test.tf-%s.com", acctest.RandString(10))
+	snippet1 := `
+	backend F_httpbin_org {
+		.always_use_host_header = true;
+		.between_bytes_timeout = 1s;
+		.connect_timeout = 1s;
+		.dynamic = true;
+		.first_byte_timeout = 1s;
+		.host = "httpbin.org";
+		.host_header = "httpbin.org";
+		.max_connections = 200;
+		.port = "443";
+		.share_key = "foo";
+		.ssl = true;
+		.ssl_cert_hostname = "httpbin.org";
+		.ssl_check_cert = always;
+		.ssl_sni_hostname = "httpbin.org";
+		.probe = {
+			.dummy = true;
+			.initial = 5;
+			.request = "HEAD / HTTP/1.1"  "Host: httpbin.org" "Connection: close";
+			.threshold = 1;
+			.timeout = 2s;
+			.window = 5;
+		  }
+	}
+	`
+	snippet2 := "# just a comment line"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviders,
+		CheckDestroy:      testAccCheckServiceV1Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceV1Config_staticBackend(name, domain, snippet1),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceV1Attributes_backends(&service, name, []string{}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "active_version", "1"),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "backend.#", "0"),
+				),
+			},
+			{
+				Config:      testAccServiceV1Config_staticBackend(name, domain, snippet2),
+				ExpectError: regexp.MustCompile("No backends"),
```

Comment:
Why do we expect an error here? If a backend is optional with a VCL service, then wouldn't having no backend defined inside the custom VCL not matter?

Reply:
@Integralist The backend "UI" object is optional, to be exact. Our compiler will definitely check if there's at least one `backend {...}` definition and throw an error if not.

Reply:
@smaeda-ks ah ok, so as long as the user has a backend defined within the VCL snippet then the VCL compiler won't complain.